core/ledger/ledger.go

```diff
@@ -207,6 +207,9 @@ func (ledger *Ledger) GetStateRangeScanIterator(chaincodeID string, startKey str
 
 // SetState sets state to given value for chaincodeID and key. Does not immideatly writes to DB
 func (ledger *Ledger) SetState(chaincodeID string, key string, value []byte) error {
+	if key == "" || value == nil {
```

Comment:
This was never supported and tested. I have added the check explicitly here to avoid issues down stream. We can add support for these later but that requires changes down stream.